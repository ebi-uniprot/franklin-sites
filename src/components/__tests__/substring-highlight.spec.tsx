import { render, screen } from '@testing-library/react';
import { SubstringHighlight } from '..';

describe('SubstringHighlight', () => {
  const string = 'Item 1 / Item 1a';
  it('should highlight substring (case insensitive)', () => {
    render(<SubstringHighlight string={string} substring="item 1" />);
    expect(screen.queryByText(string)).not.toBeInTheDocument();
    expect(screen.getByText('Item 1')).toHaveClass('highlight');
  });
  it('should render string without highlight if no substring found', () => {
    render(<SubstringHighlight string={string} substring="zap" />);
    expect(screen.getByText(string)).not.toHaveClass('highlight');
  });
});
