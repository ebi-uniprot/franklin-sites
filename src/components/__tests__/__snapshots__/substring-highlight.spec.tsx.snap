// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubstringHighlight should highlight substring (case insensitive 1`] = `
<DocumentFragment>
  Item 1 / 
  <span
    class="highlight"
  >
    Item 1
  </span>
  a
</DocumentFragment>
`;
